tensorrt_llm/_torch/auto_deploy/transformations/library/sharding.py

```diff
@@ -129,12 +123,101 @@ class BMMShardingInfo(ShardingTransformInfo):
 
     rank: int
     world_size: int
+    start_idx: int
+    end_idx: int
+
+    def validate(self, gm: GraphModule = None, node: Node = None) -> bool:
+        """Validate the transformation configuration."""
+        if not is_op(node, torch.ops.aten.bmm):
+            ad_logger.warning(f"BMM sharding is only supported for BMM nodes. Skipping {self}.")
+            return False
+
+        # Get the input tensors
+        lhs_tensor = node.args[0]
+        rhs_tensor = node.args[1]
+
+        # Check batch sizes from meta information
+        lhs_batch_size = lhs_tensor.meta["val"].shape[0]
+        rhs_batch_size = rhs_tensor.meta["val"].shape[0]
+
+        assert lhs_batch_size == rhs_batch_size, "Batch sizes of both tensors must match"
+        bmm_batch_size = lhs_batch_size
+
+        # Check if the distribution is balanced
+        remainder = bmm_batch_size % self.world_size
+
+        # NOTE: our torch.ops.auto_deploy.torch_dist_all_gather doesn't support uneven splits at the moment.
+        if remainder:
+            ad_logger.warning(
+                f"BMM batch size {bmm_batch_size} is not divisible by world size {self.world_size}. "
+                f"This will result in uneven distribution of work across devices. Skipping."
```

Comment:
Nit: technically speaking an uneven distribution is still better than no sharding at all. 

Like mentioned in line 149 it's more of an issue with our all_gather operation at the moment